python/cuml/test/test_dbscan.py

```diff
@@ -28,18 +28,20 @@
                  'no_structure']
 
 
+@pytest.mark.parametrize('max_elems_per_batch', [10, 20, 200, 2e6])
```

Comment:
Are these batch sizes all greater than the size of X (which looks like 6 rows)? I mean, does X ever get split up into multiple batches in this test? Would be nice to try an example with small max_elems that doesn't divide cleanly into the X size.

Reply:
The max elems per batch is a little confusing as it’s only indirectly controlling the batch size. This is a parameter that most users wouldn’t be expected to change. The algorithm is batched by taking the maximum number of rows of the pairwise distance matrix as possible that does not exceed the max elems per batch setting. For instance, if I have n_samples = 6 and max elems set to 20, i end up with 2 batches of 3 elements because 3*6 == 18. The reason this needed to change (and be configurable) is that the original hardcoded setting was above 16gb and would not work on 16gb devices.

Reply:
Having to explain this value in this PR confirms for me that I don’t think this approach is correct. This might make more sense to be “max memory per batch”, where we can calculate the footprint based on sizeof(T). What do you think about this?